draft-ietf-quic-multipath.md

```diff
@@ -1227,11 +1325,153 @@ PATH_AVAILABLE frame is considered lost, the peer SHOULD resend the frame
+
+
+## MP_RETIRE_CONNECTION_ID frames {#mp-retire-conn-id-frame}
+
+An endpoint sends a MP_RETIRE_CONNECTION_ID frame (type=0x15228c0a) instead of 
+RETIRE_CONNECTION_ID frame to indicate that it will no longer use a connection ID 
+that was issued by its peer. This includes the connection ID provided during the handshake. 
+Sending a MP_RETIRE_CONNECTION_ID frame also serves as a request to the peer 
+to send additional connection IDs for future use, unless the path specified 
+by Path Identifier has been abandoned. New connection IDs can be 
+delivered to a peer using the MP_NEW_CONNECTION_ID frame ({{mp-new-conn-id-frame}}).
+
+Retiring a connection ID invalidates the stateless reset token associated with that connection ID.
+
+MP_RETIRE_CONNECTION_ID frames are formatted as shown in {{fig-mp-retire-connection-id-frame-format}}.
+
+~~~
+MP_RETIRE_CONNECTION_ID Frame {
+  Type (i) = 0x15228c0a,
+  Path Identifier (i),
+  Sequence Number (i),
+}
+~~~
+{: #fig-mp-retire-connection-id-frame-format title="MP_RETIRE_CONNECTION_ID Frame Format"}
+
+Path Identifier:
+: A path identifier which is pre allocated when the Connection ID is generated, which
+means the current Connection ID can only be used on the corresponding path.
+
+Sequence Number:
+The sequence number assigned to the connection ID by the sender on the path 
+specified by Path Identifier, encoded as a variable-length integer. 
+
+
+## MAX_PATHS frames {#max-paths-frame}
+
+A MAX_PATHS frame (type=0x15228c0b) informs the peer of the cumulative number of paths 
+it is permitted to open. 
+
+MAX_PATHS frames are formatted as shown in {{fig-max-paths-frame-format}}.
+
+~~~
+MAX_PATHS Frame {
+  Type (i) = 0x15228c0b,
+  Maximum Paths (i),
+}
+~~~
+{: #fig-max-paths-frame-format title="MAX_PATHS Frame Format"}
+
+MAX_PATHS frames contain the following field:
+
+Maximum Path Identifier:
+: A count of the cumulative number of path that can be opened 
+over the lifetime of the connection. This value cannot exceed 2^32-1, as it is not 
+possible to encode Path IDs larger than 2^32-1. Receipt of a frame that permits 
+opening of a path with Path Identifier larger than this limit MUST be treated 
+as a connection error of type FRAME_ENCODING_ERROR.
+
+Loss or reordering can cause an endpoint to receive a MAX_PATHS frame with 
+a lower path limit than was previously received. MAX_PATHS frames that 
+do not increase the path limit MUST be ignored.
+
+An endpoint MUST NOT open more paths than permitted by the current path limit 
+set by its peer. For instance, a server that receives a path limit of 3 
+is permitted to open paths 0, 1, 2, but not path 3. An endpoint MUST terminate 
+a connection with an error of type MP_PROTOCOL_VIOLATION if a peer opens 
+more paths than was permitted. 
+
+Note that these frames do not describe the number of paths that can be opened 
+concurrently. The limit includes paths that have been abandoned as well as those 
+that are open.
```

Comment:
I believe this is more succinct and clearer:
```suggestion
An endpoint MUST NOT initiate a path with a path ID higher than the Maximum Paths value.
An endpoint MUST terminate the a connection with an error of type MP_PROTOCOL_VIOLATION if a peer opens more paths than was permitted. 
```

Reply:
I think it should be "An endpoint MUST NOT send an MP_NEW_CONNECTION_ID frame with a path ID higher than the Maximum Paths value".